IM_calculation/IM/im_calculation.py

```diff
@@ -119,20 +118,21 @@ def compute_adv_measure(waveform, advanced_im_config, output_dir):
     :param output_dir: Directory where output folders are contained. Structure is /path/to/output_dir/station/im_name
     :return:
     """
-    try:
-        if advanced_im_config.IM_list is not None:
-            waveform_acc = waveform[0]
-            station_name = waveform_acc.station_name
-            adv_im_out_dir = os.path.join(output_dir, station_name)
+    
+    if advanced_im_config.IM_list is not None:
+        waveform_acc = waveform[0]
+        station_name = waveform_acc.station_name
+        adv_im_out_dir = os.path.join(output_dir, station_name)
+        try:
```

Comment:
Why did you more the try? I think it was there for a reason but I can't remember.

Reply:
Is there a case IM_list is not None, but still throws an error? If not can remove the try

Reply:
@jasonmotha move or remove?

Reply:
I mean the try was meant to be checking that IM_list existed under advanced_im_config. (that's at least the error that's printed out)

Reply:
in theory, it shouldn't, if the config is passed all the way from adv_im_factory.
but if for some reason this function is called in non-typical way, it might.

Reply:
Oh, so the start of the function should have something like this?
try: if advanced_im_config.IM_list is None: print("Nothing to do"); return
except: print("Broken"); return
else: do the thing

Reply:
Kinda. I was suggesting like Jonney's current change.

Reply:
it should break the whole script, since which ever calling this function should have a config.
i've updated it to use if else, its probably easier that way.